velox/connectors/hive/storage_adapters/s3fs/S3Util.cpp

```diff
@@ -43,4 +44,107 @@ std::string getErrorStringFromS3Error(
   }
 }
 
+/// The noProxyList is a comma separated list of subdomains or domains.
+/// For a given hostname check if it has a matching domain or subdomain in
+/// the noProxyList.
+bool isHostExcludedFromProxy(
+    const std::string& hostname,
+    const std::string& noProxyList) {
+  std::vector<std::string> noProxyListElements{};
+
+  if (noProxyList.empty()) {
+    return false;
+  }
+
+  folly::split(',', noProxyList, noProxyListElements);
+  // An exact match or subdomain match is needed.
+  for (auto elem : noProxyListElements) {
+    if (elem.length() < hostname.length() && elem[0] == '.' &&
+        !hostname.compare(
+            hostname.length() - elem.length(), elem.length(), elem)) {
+      return true;
+    } else if (
+        elem.length() < hostname.length() && elem[0] == '*' && elem[1] == '.' &&
+        !hostname.compare(
+            hostname.length() - elem.length() + 1,
+            elem.length() - 1,
+            elem.substr(1))) {
+      return true;
+    } else if (elem.length() == hostname.length() && !hostname.compare(elem)) {
+      return true;
+    }
+  }
+  return false;
+}
+
+/// Reading the various proxy related environment variables.
+/// There is a lacking standard. The environment variables can be
+/// defined lower case or upper case. The lower case values are checked
+/// first and, if set, returned, therefore taking precendence.
+namespace {
```

Comment:
@kgpai says we should stop using anonymous namespaces.

Reply:
I found the discussion around this here https://github.com/facebookincubator/velox/pull/7949#issuecomment-1907461540
I guess the recommendation is to only avoid anonymous namespaces in the headers and does not apply to the cpp files. Sorry for the confusion.